### PR TITLE
refactor: replace moment.utc.toISOString in AnnotationTimeInput

### LINKED ISSUE #2029: Moment Refactor: remove moment from AnnotationTimeInput.tsx
We need to replace the instances of `moment.toDate` and `moment.toISOString` in `AnnotationTimeInput.tsx`

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #2029

part of : #1844

This PR removes `moment` from `src/annotations/components/annotationForm/AnnotationTimeInput.tsx`

### CODE REVIEW

File: src/annotations/components/annotationForm/AnnotationTimeInput.tsx [hunk 1]
```diff
@@ -77,20 +79,11 @@ export const AnnotationTimeInput: FC<Props> = (props: Props) => {
 
     if (isValidTimeFormat(event.target.value)) {
       if (timeZone === 'UTC') {
-        props.onChange(
-          moment
-            .utc(event.target.value, timeFormat)
-            .toDate()
-            .toISOString()
-        )
+        props.onChange(setTimeToUTC(event.target.value))
```
Comment (Asalem1) [inline src/annotations/components/annotationForm/AnnotationTimeInput.tsx:82]: why not do the same thing you're doing below?
```suggestion
        props.onChange(new Date(event.target.value).toISOString())
```

Comment (ChitlangeSahas) [reply src/annotations/components/annotationForm/AnnotationTimeInput.tsx:82]: We'd need a UTC value, for the time value passed. Hence. If that makes sense.

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 62 | :yellow_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2048)

File: src/annotations/components/annotationForm/AnnotationTimeInput.tsx (hunk 1 above)
Comment (Asalem1) [reply src/annotations/components/annotationForm/AnnotationTimeInput.tsx:81]: K, is it a selector or a utility function that exists within a selector file?

Comment (ChitlangeSahas) [reply src/annotations/components/annotationForm/AnnotationTimeInput.tsx:81]: Yah! its in `src/dashboards/selectors`

Comment (ChitlangeSahas) [reply src/annotations/components/annotationForm/AnnotationTimeInput.tsx:81]: it seems like a utility function. Needed in the selectors.

Comment (Asalem1) [reply src/annotations/components/annotationForm/AnnotationTimeInput.tsx:81]: Gotcha.

File: src/shared/utils/dateTimeUtils.ts [hunk 2]
```diff
@@ -42,3 +42,24 @@ export function addDurationToDate(
...
+
+  if (hours === '12') {
+    hours = '00'
+  }
+
+  if (modifier === 'PM') {
+    hours = parseInt(hours, 10) + 12
+  }
+
+  return `${date} ${hours}:${minutes}:${seconds}`
```
Comment (ChitlangeSahas) [inline src/shared/utils/dateTimeUtils.ts:64]: I'll add unit tests for this, I just want to see if my approach is correct.

**Pull request closed without merging**